### ISSUE TITLE
Incorporate VileR's normal 4-6x scaler patch

### ISSUE THREAD

**Issue by James-F2** [dreamer/dosbox-staging#152]
https://www.vogons.org/viewtopic.php?f=41&t=43899
This patch is mandatory imo.

320x200 games x3 result in only 600 vertical resolution which is scaled to Desktop resolution and it is blurry.
I intentionally use the interpolated options like opengl with higher scaling factor to have properly interpolated image and maintain sharp pixels with correct aspect ratio,, this cannot be done with NB option which distorts the pixels.

**Comment by dreamer** [comment]
Patch itself: e2c368de27b69e3d1d91dd60744ae44aa256b185 (we have it on our "branch mirror" of ECE).

I haven't used this patch myself yet, but I find this quite important also for resolutions higher than 1920x1080 - such displays are already popular. This is a completely separate issue from pixel-perfect improvements, that @ant-222 is working on.

There is slight incompatibility with the latest commits, that landed in SVN, namely: ec8b59ab06641030ddb1b44dba9c96153a0db2a4 and 6d11dd5eac7b2ce8a2cb2608dc8fbdf562c94179, and I remember some comments about possible improvements from Qbix in the Vogons thread, so it'll require a bit of work - hopefully not too much.

**Comment by ant-222** [comment]
@dreamer:
>  This is a completely separate issue from pixel-perfect improvements, that **@ant-222** is working on.
What use-cases of the normal scalers are not covered by the pixel-perfect mode? The ability to magnify beyond the screen dimensions and then downscale interpolatively with minimal blur? This should be possible with an *SDL* hardware surface. I can implement such scaling after pixel-perfect.

**Comment by James-F2** [comment]
It should behave exactly like your surfacenp (near-perfect) option.
But, I remember that the surfacenp has some stuttering during fade-in/out scenes, hence I resorted to using Scale x6 and Opengl.

**Comment by ant-222** [comment]
Understood, James. Whereas a pro might try to explain why the hard-coded scalers are so much faster than my patch, I think the right solution would be to let the hardware to the hard work, and software the soft. I can return to this only after I a working pixel-perfect mode is accepted into `master`.

**Comment by James-F2** [comment]
Thank you.
That would be a much better solution than fixed 4-6.

**Comment by ant-222** [comment]
I will to implement a fast `texturenp'.

**Comment by James-F2** [comment]
As of svn [4319] dosbox support opengl shaders,, and a "sharp" filter does exactly what I want or surfacenp does.
Issue can be closed?

**Comment by dreamer** [comment]
r4319 is (massively) conflicting with SDL2. I am in the middle of conflict resolution right now - once done and verified working, I will close this issue (but it may take a while).

**Comment by dreamer** [comment]
r4319 was just merged to master branch.
